scipy/interpolate/_aaa.py

```diff
@@ -309,6 +373,77 @@ def _compute_weights(self, f, z, rtol, max_terms):
         self.support_values = fj[i_non_zero]
         self.weights = wj[i_non_zero]
         self.errors = errors[: m + 1]
+        self._points = z
+        self._values = f
+
+    def clean_up(self, cleanup_tol=1e-13):
+        """Automatic removal of Froissart doublets.
+
+        Parameters
+        ----------
+        cleanup_tol : float, optional
+            Poles with residues less than this number times the geometric mean
+            of `values` times the minimum distance to `points` are deemed spurious by
+            the cleanup procedure, defaults to 1e-13.
+
+        Returns
+        -------
+        int
+            Number of Froissart doublets detected
+        """
+        # Find negligible residues
+        geom_mean_abs_f = scipy.stats.gmean(np.abs(self._values))
+
+        Z_distances = np.min(
+            np.abs(np.subtract.outer(self.poles(), self._points)), axis=1
+        )
+
+        with np.errstate(divide="ignore", invalid="ignore"):
+            ii = np.nonzero(
+                np.abs(self.residues()) / Z_distances < cleanup_tol * geom_mean_abs_f
+            )
+
+        ni = ii[0].size
+        if ni == 0:
+            return
```

Comment:
```suggestion
            return ni
```
?

Reply:
Oops yes thanks, brain clearly not fully switched on